### ISSUE TITLE
Feature Request: Allow generation and multiple use of Mapper Schema

### ISSUE THREAD

**Issue by blacksun1** [midknight41/map-factory#6]
Hey,

Would be great to be able to use something like this

```js
function pointSchemaFactory() {

  const map = CreateMapper() // No source object goes in here
  map("x").to("point.x");
  map("y").to("point.y");

  return map;
}

const pointMapper = pointSchemaFactory();
assert.deepEqual(pointMapper({x: 100, y: 100}, {point: {x: 100, y: 100});
assert.deepEqual(pointMapper({x: 50, y: 100}, {point: {x: 50, y: 100});
assert.deepEqual(pointMapper({x: 50, y: 50}, {point: {x: 50, y: 50});
assert.deepEqual(pointMapper({x: 100 y: 50}, {point: {x: 100, y: 50});
```

Where the mapping is reusable rather than having a function that creates the mapping before executing it every time the source is different.

**Comment by midknight41** [comment]
```js
function pointSchemaFactory() {

  const map = CreateMapper() // No source object goes in here
  map("x").to("point.x");
  map("y").to("point.y");

  return map;
}

const map = pointSchemaFactory();
assert.deepEqual(map.execute({x: 100, y: 100}), {point: {x: 100, y: 100});
assert.deepEqual(map.execute({x: 50, y: 100}), {point: {x: 50, y: 100});
```

Basically make ```createMapper()``` build a resuable map and have ```map.execute``` take an object as a parameter?

That makes a lot more sense actually!

**Comment by midknight41** [comment]
I might have a crack at that on the way home...

**Comment by midknight41** [comment]
@blacksun1 I've done this change. I'll publish as v1 shortly.

Also tidied up the dist folder.

**Comment by midknight41** [comment]
The code in git is currently compatible with both the old and new approaches.

**Comment by blacksun1** [comment]
Snap! I wrote basically the same thing this morning. LOL

https://github.com/blacksun1/map-factory/commit/f0bc9849672664d61e43bec9874c17dc9f20c5c4

**Comment by blacksun1** [comment]
Did you write that on the train last night?

**Comment by midknight41** [comment]
Yup.

**Comment by blacksun1** [comment]
BTW, you probably need to explicitly throw if the user has not provided a sourceObject via either the constructor or the execute method.

you could just do

```js
assert(sourceObject, "You need to provide a source either on construction or in the execute")
```

**Comment by midknight41** [comment]
Yeah. I was toying with just breaking the old approach to be honest. I'm going to bump major anyway. Supplying to the object to ```execute()``` is much much better.

To approaches is just confusing.

**Comment by midknight41** [comment]
Yeah. Let's do that. No point to add the complexity to the code base.

**Comment by midknight41** [comment]
The build stuff you added is awesome by the way. Thanks!

**Comment by blacksun1** [comment]
No worries.

**Comment by midknight41** [comment]
published as 1.0.1. I mucked up something for 1.0.0. :cry:

**Comment by bharanidharan-k** [comment]
Awesome. I was looking for same thing. Define once, reuse multiple times!
